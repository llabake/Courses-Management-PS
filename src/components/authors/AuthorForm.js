import React from 'react';
import {
  string,
  shape,
  arrayOf,
  bool,
  func,
  object,
} from 'prop-types';
import TextInput from '../common/TextInput';

const AuthorForm = ({
  author,
  onSave,
  onChange,
  saving,
  errors,
}) => (
  <form>
    <h1>Manage Authors</h1>
    <TextInput
        name="firstName"
        label="First Name"
        value={author.firstName}
        onChange={onChange}
        error={errors.firstName}
      />

    <TextInput
      name="lastName"
      label="Last Name"
      value={author.lastName}
      onChange={onChange}
      error={errors.lastName}
      />

    <input
        type="submit"
        disabled={saving}
        value={saving ? 'Saving' : 'Save'}
        className="btn btn-primary"
        onClick={onSave}
      />
  </form>
);

export default AuthorForm;
