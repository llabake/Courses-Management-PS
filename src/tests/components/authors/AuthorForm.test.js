import React from 'react';
import { shallow } from 'enzyme';
import AuthorForm from "../../../components/authors/AuthorForm";

describe('<AuthorForm/>', () => {
  it('should render correctly', () => {
    const props = {
      author: {},
      onSave: jest.fn(),
      onChange: jest.fn(),
      saving: false,
      errors: [],
    };
    const wrapper = shallow(<AuthorForm {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input').props().value).toBe('Save');
  });
  it('should should saving when saving', () => {
    const props = {
      author: {},
      onSave: jest.fn(),
      onChange: jest.fn(),
      saving: true,
      errors: [],
    };
    const wrapper = shallow(<AuthorForm {...props} />);
    expect(wrapper.find('input').props().value).toBe('Saving');
  });
});
