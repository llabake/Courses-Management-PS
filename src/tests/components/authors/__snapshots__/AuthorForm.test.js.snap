// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorForm/> should render correctly 1`] = `
<form>
  <h1>
    Manage Authors
  </h1>
  <TextInput
    label="First Name"
    name="firstName"
    onChange={[MockFunction]}
  />
  <TextInput
    label="Last Name"
    name="lastName"
    onChange={[MockFunction]}
  />
  <input
    className="btn btn-primary"
    disabled={false}
    onClick={[MockFunction]}
    type="submit"
    value="Save"
  />
</form>
`;
